interfaces/kmod/backend_test.go

```diff
@@ -117,11 +98,9 @@ func (s *backendSuite) TestInstallingSnapCreatesModulesConf(c *C) {
 
 func (s *backendSuite) TestRemovingSnapRemovesModulesConf(c *C) {
 	// NOTE: Hand out a permanent snippet so that .conf file is generated.
-	s.Iface.PermanentSlotSnippetCallback = func(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
-		if securitySystem == interfaces.SecurityKMod {
-			return []byte("module1\nmodule2"), nil
-		}
-		return nil, nil
+	s.Iface.KModPermanentSlotCallback = func(spec *kmod.Specification, slot *interfaces.Slot) error {
+		spec.AddModule("module1")
```

Comment:
This callback is irrelevant for this test case, it's there just to have a non-empty list of modules, but what we're really testing is that the generated file is removed when snap is uninstalled.